### PR TITLE
Make Response parser more relaxed on reason message #981

### PR DESCRIPTION
**Pull request by athieriot**
Proposal for issue #981 that will allow to parse a non-standard
Http Response which does not contain a status message.

Example: 'HTTP/1.1 400' instead of 'HTTP/1.1 400 Bad Request'

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala [hunk 1]
```diff
@@ -85,19 +85,26 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
         }
       }
     }
+
+    def isNewLine(idx: Int) = byteChar(input, idx) == '\n' || byteChar(input, idx) == '\r' && byteChar(input, idx + 1) == '\n'
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala]: Can you extract the other two conditions as well? 

```scala
def isLF ...
def isCRLF ...
def isNewLine ...

def skipNewLine(idx: Int) = 
  if (isCRLF(idx)) idx + 2
  else if (isLF(idx)) idx + 1
  else throw new IllegalStateException("Newline expected")
```

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala [hunk 2]
```diff
@@ -85,19 +85,26 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
...
-          if (byteChar(input, idx) == '\r' && byteChar(input, idx + 1) == '\n') idx + 2
-          else if (byteChar(input, idx) == '\n') idx + 1
+          if (isNewLine(idx)) skipNewLine(idx)
           else skipReason(idx + 1)
         else throw new ParsingException("Response reason phrase exceeds the configured limit of " +
           maxResponseReasonLength + " characters")
       skipReason(startIdx)
-    } else if (byteChar(input, cursor + 3) == '\n' || byteChar(input, cursor + 3) == '\r' && byteChar(input, cursor + 4) == '\n') {
-      throw new ParsingException("Status code misses trailing space")
+    } else if (isNewLine(cursor + 3)) {
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala]: Add a comment here that we accept this format as well diverging from the spec. Link to the issue.

**Comment by athieriot** [comment]
Thanks for the review. I won't have the time today but will definitely update it soon.

**Comment by jrudolph** [comment]
That's fine! Thanks again.

**Comment by athieriot** [comment]
Updated

**Comment by athieriot** [comment]
Oh noes...

**Comment by jonas** [comment]
The failing test is #299

**Comment by jonas** [comment]
PLS BUILD

**Comment by jonas** [comment]
Another ConnectionPoolSpec failure: The host-level client infrastructure should respect the configured `maxRetries` value.

 - https://jenkins.akka.io:8498/job/pr-validator-akka-http/912/

**Comment by jonas** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
(Sorry for the useless merge to master and the failing test. Not related to this PR, will rebuild once the compilation on master has been fixed)

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by ktoso** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
Thanks, @athieriot!

**Comment by athieriot** [comment]
Thank you :)

**Pull request merged**